include/zephyr/posix/pthread.h

```diff
@@ -366,10 +373,14 @@ int pthread_mutexattr_getrobust(const pthread_mutexattr_t * int *);
 int pthread_mutexattr_setprioceiling(pthread_mutexattr_t *, int);
 int pthread_mutexattr_setpshared(pthread_mutexattr_t *, int);
 int pthread_mutexattr_setrobust(pthread_mutexattr_t *, int);
-int pthread_barrierattr_getpshared(const pthread_barrierattr_t *, int *);
-int pthread_barrierattr_setpshared(pthread_barrierattr_t *, int);
 */
 
+int pthread_barrierattr_destroy(pthread_barrierattr_t *attr);
+int pthread_barrierattr_init(pthread_barrierattr_t *attr);
+int pthread_barrierattr_setpshared(pthread_barrierattr_t *attr, int pshared);
+int pthread_barrierattr_getpshared(const pthread_barrierattr_t *restrict attr,
+				   int *restrict pshared);
+
```

Comment:
Some of these are already defined above static inline.

Can you also please remove these, convert them to not-static-inline, and lastly declare `pthread_barrierattr_setpshared()` and `pthread_barrierattr_getpshared()` in a similar manner?